test/python/fx_importer/basic_test.py

```diff
@@ -0,0 +1,80 @@
+# Copyright 2023 Advanced Micro Devices, Inc
+#
+# Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+# Also available under a BSD-style license. See LICENSE.
+
+# RUN: %PYTHON %s | FileCheck %s
+
+from typing import Optional
+
+import torch
+import torch.export
+import torch.nn as nn
+
+from torch_mlir.extras.fx_importer import FxImporter
+from torch_mlir import ir
+from torch_mlir.dialects import torch as torch_d
+
+
+def export_and_import(
+    f,
+    *args,
+    fx_importer: Optional[FxImporter] = None,
+    constraints: Optional[torch.export.Constraint] = None,
+    **kwargs,
+):
+    context = ir.Context()
+    torch_d.register_dialect(context)
+
+    if fx_importer is None:
+        fx_importer = FxImporter(context=context)
+    prog = torch.export.export(f, args, kwargs, constraints=constraints)
```

Comment:
Nice, I was wondering if we'd have to take a `shark_turbine.aot` dependency (i.e. `export` or `CompiledModule` API) to get the dynamo exported program, but looks like not :)

Just for my understanding, could you clarify/document what's different between the native `torch.export.export` in comparison to Turbine's AOT exporter, in terms of the constraints/knobs they expose (e.g. [dynamic dim constraints](https://pytorch.org/docs/stable/export.html#torch.export.dynamic_dim)). In other words, do you anticipate relying more directly on the `torch.export` path going forward for all of torch-mlir's use-cases?

Reply:
The turbine side adds a lot for assembling multi graph programs and having options to exploit iree features. But it does all of this via hooks into the fx_importer when needed.

Turbine uses a predecessor API to the now public torch `ExportedProgram` and will eventually upgrade to the same thing as here under the hood. The main difference is in how tensors from "outside" are handled. The predecessor API just let them flow through and the importer has to do a bit of work to resolve them. The`ExportedProgram` API does the extra accounting at the FX level. It also looks like it has the beginnings of new features like call graphs. Mainly a v0/V1 difference if I read it properly. Figured I'd just start upstream on the V1 thing.